addons/udes_stock/tests/test_refactoring_splitting.py

```diff
@@ -918,8 +926,129 @@ def test_split_mixed_products_partial_reserve(self):
+        self.assertEqual(self.pick_2.state, "done")
+        self.assertTrue(self.pick_2.date_done)
+        self.assertNotEqual(backorder.state, "done")
+        self.assertFalse(backorder.date_done)
+
+        putaway_picking = self.pick_2.u_next_picking_ids
+
+        # check that putaway is not in state done and does not have date done set
+        self.assertNotEqual(putaway_picking.state, "done")
+        self.assertFalse(putaway_picking.date_done)
+
+    def test_post_validate_refactor(self):
+        """Test123"""
+
+        mv = self.create_move(self.banana, 1, self.pick_1)
+        pickings = self.pick_1 | self.pick_2
+        self.pick_1.action_assign()
+        self.assertEqual(len(pickings.mapped("move_lines")), 4)
+        for move_line in pickings.mapped("move_line_ids"):
+            move_line.write(
+                {"location_dest_id": self.received_location.id, "qty_done": move_line.product_qty}
+            )
+        pickings.action_done()
+        self.assertEqual(self.pick_1.date_done, self.pick_2.date_done)
+
+        # Group by product post validation
+        self.picking_type_in.write(
+            {
+                "u_post_validate_action": "group_by_move_key",
+                "u_move_key_format": "{product_id.id}",
+            }
+        )
+        # Change the time shift of the apple move and picking time,
+        # so any new existing refactored pickings with apples have the greatest date,
+        # Or at least greater than bananas.
+        time_shift = datetime.datetime.now() + datetime.timedelta(hours=1)
+        self.pick_1.date_done = time_shift
+        self.pick_1.move_lines.filtered(lambda mv: mv.product_id == self.apple).write({"date": time_shift})
+
+        self.assertGreater(self.pick_1.date_done, self.pick_2.date_done)
+
+        # Call refactor
+        pickings.mapped("move_lines")._action_refactor(stage="validate")
+
+        all_pickings = self.env["stock.picking"].search([("picking_type_id", "=", self.picking_type_in.id)])
+        draft_pickings = all_pickings.filtered(lambda p: p.state == "draft")
+        refactored_pickings = all_pickings - draft_pickings
+        apple_picking = refactored_pickings.filtered(lambda p: p.mapped("move_lines.product_id") == self.apple)
+        banana_picking = refactored_pickings - apple_picking
+        self.assertEqual(len(apple_picking), 1)
+        self.assertEqual(len(banana_picking), 1)
+        # Due to refactoring, the pickings get moved out of a done picking
+        # So empty pickings have a date done. This should be ok provided these
+        # draft pickings are never reused - but unsure if other refactoring takes
+        # use of draft pickings. (Presumably unlink resolves this issues quickly in refactoring)
+        # self.assertEqual(draft_pickings.mapped("date_done"), [False])
+        self.assertTrue(banana_picking.date_done)
+        for move in apple_picking.move_lines:
```

Comment:
Acutally lookign back at this might want to just comapre the move dates directly

apple_mv_1, apple_mv_2 = apple_picking.move_lines.sorted(key=lambda mv: mv.date)
self.assertGreater(apple_mv_2.date, apple_mv_1.date)
self.assertEqual(apple_picking.date_done, apple_mv_2.date)

Similar for banana - but that they are all equal